hack/generate-pipelines-readme.py

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env python
+import json
+import subprocess
+import os
+import re
+import yaml
+import shutil
+from pathlib import Path
+from tempfile import mkdtemp
+
+PIPELINE_GENERATE_INPUT_DIRS = ('./pipelines/', './pipelines/rhtap/')
+PIPELINES_DIR = './pipelines/'
+TASKS_DIR = './task/'
+# mapping pipeline_name to directory name, in case it isn't the same
+PIPELINE_TO_DIRECTORY_MAPPING = {'gitops-pull-request': 'gitops-pull-request-rhtap'}
+
+
+def run(cmd):
+    print("Subprocess: %s" % ' '.join(cmd))
+    failed = 1
+
+    try:
+        process = subprocess.run(cmd, check=True, capture_output=True)
+        failed = 0
+    except subprocess.CalledProcessError as e:
+        print(f"{cmd[0]} failed:\nSTDOUT:\n{e.stdout.decode()}\nSTDERR:\n{e.stderr.decode()}")
+    except FileNotFoundError:
+        print(f"command: {cmd[0]} doesn't exist")
+        return "", "", failed
+
+    if process.stderr:
+        print(f"{cmd[0]} STDERR:\n{process.stderr.decode()}")
+
+    return process.stdout, process.stderr, failed
+
+
+def main():
+    temp_dir = mkdtemp()
+
+    for input_pipeline_dir in PIPELINE_GENERATE_INPUT_DIRS:
+        generate_pipelines_cmd = ["oc", "kustomize", "--output", temp_dir, input_pipeline_dir]
+        _, _, failed = run(generate_pipelines_cmd)
+        if failed:
+            shutil.rmtree(temp_dir)
+            exit(1)
+
+    for f in os.listdir(temp_dir):
+        pipeline_dir = f.replace("tekton.dev_v1_pipeline_", "").replace(".yaml", "")
+        if pipeline_dir.startswith("enterprise"):
+            continue
+
+        pipeline_dir = PIPELINE_TO_DIRECTORY_MAPPING.get(pipeline_dir, pipeline_dir)
+
+        full_path = Path(PIPELINES_DIR).joinpath(pipeline_dir)
+        if not full_path.exists():
+            print(f"pipeline directory: {full_path}, for pipeline: {pipeline_dir} doesn't exist")
+            shutil.rmtree(temp_dir)
+            exit(1)
+
+        pipelines_info = {}
+        with open(Path(temp_dir).joinpath(f), 'r') as f:
+            pipeline_data = yaml.safe_load(f)
+
+        if pipeline_data.get('kind') != 'Pipeline':
+            print(f"generated yaml file isn't pipeline: {f} will skip it")
+            continue
+
+        pipeline_name = pipeline_data['metadata']['name']
+        pipelines_info[pipeline_name] = {'params': [], 'results': [], 'workspaces': [], 'tasks': []}
+
+        for param in pipeline_data['spec'].get('params', []):
+            param_dict = {'used': []}
+            param_dict['name'] = param.get('name')
+            param_dict['description'] = param.get('description', "")
+            param_dict['default'] = param.get('default', None)
```

Comment:
Sometimes the `default` is shown as an empty string, sometimes as "None"

![image](https://github.com/redhat-appstudio/build-definitions/assets/38193343/7952ff12-bd9f-4050-8e96-28a25992a201)

If I understand correctly, the difference is that None means that the param is required. Can we make this more understandable for readers?

Reply:
that isn't exactly truth .....

those params without default values are set in build service

there isn't really any real concept of required params

Reply:
If a param has no default, it is required. Build-service sets the required params, but that doesn't mean they're not required

Reply:
already had discussion about 'required' with @arewm and we decided to not add it to create confusion